mesonbuild/backend/ninjabackend.py

```diff
@@ -592,9 +592,19 @@ def generate_coverage_rules(self, outfile):
 
     def generate_install(self, outfile):
         install_data_file = os.path.join(self.environment.get_scratch_dir(), 'install.dat')
+        if self.environment.is_cross_build():
```

Comment:
We also support installing `native: true` build targets, don't we? So maybe we should check for that too?

Or just disallow `install: true` on targets that are `native: true` in `BuildTarget().process_kwargs()` which seems more sane to me.